File: src/snippets/search/binary-search/binary-search.py
def binary_search(arr, target):
    """
    Perform binary search on a sorted array to find the index of the target element.

    :param arr: List of sorted elements
    :param target: Element to search for
    :return: Index of the target element if found, otherwise -1
    """
    left, right = 0, len(arr) - 1

    while left <= right:
        mid = left + (right - left) // 2

        # Check if target is present at mid
        if arr[mid] == target:
            return mid
        # If target is greater, ignore left half
        elif arr[mid] < target:
            left = mid + 1
        # If target is smaller, ignore right half
        else:
            right = mid - 1

    # Target is not present in array
    return -1

# Example usage
if __name__ == "__main__":
    arr = [2, 3, 4, 10, 40]
    target = 10

    result = binary_search(arr, target)
    if result != -1:
        print(f"Element is present at index {result}")
    else:
        print("Element is not present in array")

File: src/snippets/search/binary-search/binary-search.cs
using System;

class BinarySearchExample
{
    // Binary Search function
    public static int BinarySearch(int[] array, int target)
    {
        int left = 0;
        int right = array.Length - 1;

        while (left <= right)
        {
            int mid = left + (right - left) / 2;

            // Check if target is present at mid
            if (array[mid] == target)
                return mid;

            // If target greater, ignore left half
            if (array[mid] < target)
                left = mid + 1;
            // If target is smaller, ignore right half
            else
                right = mid - 1;
        }

        // Target is not present in array
        return -1;
    }

    // Main method to test the binary search function
    static void Main()
    {
        int[] array = { 2, 3, 4, 10, 40 };
        int target = 10;
        int result = BinarySearch(array, target);

        if (result != -1)
            Console.WriteLine("Element found at index " + result);
        else
            Console.WriteLine("Element not present in array");
    }
}